channelserver/README.md

```diff
@@ -29,10 +29,35 @@ $ cargo run
 
 ## API
 
-When connecting to the server as a new session, the first response
-message contains the URI path to send to the counterpart client, This
-is known as the "channel". Currently channels are limited by the
-`max_clients` config option to 2 sessions.
+When connecting to the server as a new session, the first response message contains a JSON response containingthe URI path to send to the counterpart client, this is known as the "channel", and the discrete channelID.
+
+e.g. for a connection to `wss://example.com/v1/ws/`
+```json
+{"channelid":"IZ5B8Wj2qR1NlsNbSXQ2Fg","link":"/v1/ws/IZ5B8Wj2qR1NlsNbSXQ2Fg"}
+```
+Additional connections can be made to the URI specified in `link`.
+
+Messages sent are expected to be URL Safe base64 encoded blocks and are delivered wrapped in a JSON envelope containing the message and sender meta data.
+
+e.g.
+```json
+{"message":"aBc12e....","sender":{"city":"Mountain View","country":"USA","region":"California","remote":"10.0.0.1", }}
+```
+
+This will attempt to localize based on the preferred `Accept-Languages:` HTTP header. If no header is provided, results are unspecified (although probably in German). If an aspect of the location cannot be determined, it is not included in the output.
```

Comment:
```suggestion
This will attempt to localize the geolocation data based on the preferred `Accept-Languages:` HTTP header. If no header is provided, results are unspecified (although probably in German). If an aspect of the location cannot be determined, it is not included in the output.
```